aten/src/ATen/core/IListRef_test.cpp

```diff
@@ -27,12 +27,16 @@ static std::vector<optional<at::Tensor>> get_boxed_opt_tensor_vector() {
 }
 
 static std::vector<at::OptionalTensorRef> get_unboxed_opt_tensor_vector() {
+  static std::vector<at::Tensor> tensors;
   std::vector<at::OptionalTensorRef> optional_tensors;
   const size_t SIZE = 5;
   for (size_t i = 0; i < SIZE * 2; i++) {
-    auto opt_tensor = (i % 2 == 0) ? at::OptionalTensorRef(at::empty({0}))
-                                   : at::OptionalTensorRef();
-    optional_tensors.emplace_back(opt_tensor);
+    if (i % 2 == 0) {
+      tensors.push_back(at::empty({0}));
```

Comment:
Won't this make the tensors std::vector  just grow infinitely if this function is called over and over again or am I missing something?

Reply:
No, it is a private function that called only once for a test unit and there is no danger for growing  out of memory.